editor/plugins/material_editor_plugin.cpp

```diff
@@ -151,19 +158,38 @@ void MaterialEditor::_on_light_2_switch_pressed() {
 }
 
 void MaterialEditor::_on_sphere_switch_pressed() {
-	box_instance->hide();
 	sphere_instance->show();
-	box_switch->set_pressed(false);
+	box_instance->hide();
+	quad_instance->hide();
 	sphere_switch->set_pressed(true);
-	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_on_sphere", true);
+	box_switch->set_pressed(false);
+	quad_switch->set_pressed(false);
+	_set_rotation(-15.0, 30.0);
+	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_shape", "sphere");
```

Comment:
I don't think any code is needed to deal with the previous boolean project metadata here? The only effect should be the default preview mesh changing once, the first time you load the editor with this new option.

Should these be `SNAME()` or an enum, though? I didn't see any example of string project metadata exactly like this.